prow/config/jobs.go

```diff
@@ -92,8 +93,10 @@ type JobBase struct {
 	Cluster string `json:"cluster,omitempty"`
 	// SourcePath contains the path where this job is defined
 	SourcePath string `json:"-"`
-	// Spec is the Kubernetes pod spec used if Agent is Kubernetes.
+	// Spec is the Kubernetes pod spec used if Agent is kubernetes.
 	Spec *v1.PodSpec `json:"spec,omitempty"`
+	// BuildSpec is the Knative build spec used if Agent is knative-build.
+	BuildSpec *buildv1alpha1.BuildSpec `json:"build_spec,omitempty"`
```

Comment:
Is adding the `BuildSpec` field related to what the rest of this PR is doing?

Reply:
seems like this ought to be a separate commit for adding the buildspec? the PR is small enough that I don't think it needs to be a different PR necessarily, but it might help with the commit history to break these up.

Reply:
Done